Templates/make_jenkins_EC2-Agent-Windows-instance.tmplt.json

```diff
@@ -0,0 +1,1613 @@
+            "AllowedPattern": "^$|^s3://(.*)/(.*)\\.(bat|ps1)$",
+            "ConstraintDescription": "Must use an S3 URL (starts with \"s3://\") to a .bat or .ps1 file",
+            "Default": "",
+            "Description": "(Optional) S3 URL to the .ps1 or .bat application script in an S3 bucket (s3://). Leave blank to launch without an application script. If specified, an appropriate \"InstanceRole\" is required",
+            "Type": "String"
+        },
+        "AppVolumeDevice": {
+            "AllowedValues": [
+                "",
+                "xvdf",
+                "xvde",
+                "xvdg",
+                "xvdh",
+                "xvdi"
+            ],
+            "Default": "",
+            "Description": "(Optional) Device to mount an extra EBS volume. Leave blank to launch without an extra application volume",
+            "Type": "String"
+        },
+        "AppVolumeSize": {
+            "ConstraintDescription": "Must be between 1GB and 16384GB.",
+            "Default": "1",
+            "Description": "Size in GB of the EBS volume to create. Ignored if \"AppVolumeDevice\" is blank",
+            "MaxValue": "16384",
+            "MinValue": "1",
+            "Type": "Number"
+        },
+        "AppVolumeType": {
+            "AllowedValues": [
+                "gp2",
+                "io1",
+                "sc1",
+                "st1",
+                "standard"
+            ],
+            "Default": "gp2",
+            "Description": "Type of EBS volume to create. Ignored if \"AppVolumeDevice\" is blank",
+            "Type": "String"
+        },
+        "CfnEndpointUrl": {
+            "AllowedPattern": "^$|^http[s]?://.*$",
+            "Default": "https://cloudformation.us-east-1.amazonaws.com",
+            "Description": "(Optional) URL to the CloudFormation Endpoint. e.g. https://cloudformation.us-east-1.amazonaws.com",
+            "Type": "String"
+        },
+        "ChainScriptUrl": {
+          "AllowedPattern": "^$|^http[s]?://.*$|s3://.*$",
+          "Default": "",
+          "Description": "(Optional) URL to a \"chain\" script - a script to install add-ons not part of the core agent install.",
+          "Type": "String"
+        }, 
+        "ChainScriptParams": {
+          "Default": "",
+          "Description": "(Optional) Params to pass to chain load.",
+          "Type": "String"
+        }, 
+        "CloudWatchAgentUrl": {
+            "AllowedPattern": "^$|^s3://.*$",
+            "Default": "",
+            "Description": "(Optional) S3 URL to CloudWatch Agent installer. Example: s3://amazoncloudwatch-agent/windows/amd64/latest/AmazonCloudWatchAgent.zip",
+            "Type": "String"
+        },
+        "InstanceRole": {
+            "Default": "",
+            "Description": "(Optional) IAM instance role to apply to the instance(s)",
+            "Type": "String"
+        },
+        "InstanceType": {
+            "AllowedValues": [
+                "t2.micro",
```

Comment:
`.micro` (and `.nano`) types generally don't have enough memory to support running the Java-based agent process. Ensure that your list of types only includes sizes that have at least 2GiB of RAM - or your deployed instances won't have enough memory to allocate to the OS _and_ the Jenkins JVM(s).